### PR TITLE
Enhancement/users list last login

### PR DESCRIPTION
**Pull request by SteRiccio**

### CODE REVIEW

File: server/modules/user/repository/userRepository.js [hunk 1]
```diff
@@ -75,23 +75,45 @@ export const countUsersBySurveyId = async (surveyId, countSystemAdmins = false,
...
+  const orderBy = orderByFieldBySortBy[sortBy] || 'email'
+  const orderByDirection = sortOrder && sortOrder.toUpperCase() === 'ASC' ? 'ASC' : 'DESC'
+
+  return `
+    WITH us AS (
+      SELECT DISTINCT ON (us.sess #>> '{passport,user}')
+        (us.sess #>> '{passport,user}')::uuid AS user_uuid,
+        (us.expire - interval '30 days') AS last_login_time
+      FROM user_sessions us
+      WHERE us.sess #>> '{passport,user}' IS NOT NULL
```
Comment (ramirobg94) [inline server/modules/user/repository/userRepository.js:95]: Maybe session is better than `sess`?

File: server/modules/user/repository/userRepository.js [hunk 2]
```diff
@@ -75,23 +75,45 @@ export const countUsersBySurveyId = async (surveyId, countSystemAdmins = false,
...
-export const fetchUsers = async ({ offset = 0, limit = null }, client = db) =>
-  client.map(
-    `
-    SELECT ${selectFieldsCommaSep}
+const _usersSelectQuery = ({ selectFields, sortBy = 'email', sortOrder = 'ASC' }) => {
+  // check sort by parameters
+  const orderByFieldBySortBy = {
+    email: 'email',
+    name: 'name',
+    last_login_time: 'status, last_login_time', // 'status' is used to group users that have accepted invitation, otherwise they have never logged in
```
Comment (ramirobg94) [inline server/modules/user/repository/userRepository.js:83]: should we use the keys, i think that are repeated between files

File: server/modules/user/repository/userRepository.js (hunk 1 above)
Comment (SteRiccio) [reply server/modules/user/repository/userRepository.js:95]: eheheh user_sessions is the table used by node-connect-pg-simple... we cannot rename the columns I think.

File: server/modules/user/repository/userRepository.js (hunk 2 above)
Comment (SteRiccio) [reply server/modules/user/repository/userRepository.js:83]: uhm, these are not the keys but the column names... and in some cases they are the same. Should we introduce an object with the column names?

Comment (ramirobg94) [reply server/modules/user/repository/userRepository.js:83]: yep Let's keep at it is!

**Review by ramirobg94 — APPROVED** [review]

**Pull request merged**